tests/models/encoder_decoders/wikitables_semantic_parser_test.py

```diff
@@ -58,47 +59,60 @@ def test_all_types_present_in_type_vector(self):
         # representing its index will be in x
         assert x.size(0) * x.size(1) == 2 * sums.size(0) * sums.size(1)
 
-    def test_get_prob_entity_valid_probability_distribution(self):
+    def test_get_linking_probabilities_valid_probability_distribution(self):
```

Comment:
This test is still really long and complicated.  You want to have as little logic in a test case as possible.  For example, see some of these tests:
https://github.com/allenai/allennlp/blob/4f8e04e6cdfbc8707707ac978a964777459c161e/tests/models/encoder_decoders/wikitables_semantic_parser_test.py#L316-L337
This test has about 20 lines, but a lot of those are comments, and there is no complex logic.  We're just instantiating some objects, calling a method, and asserting that the return value is what we expect.  The more logic you have in your test, the harder it is to understand, and the easier it is to not actually be testing what you think you're testing.

An example of what I meant by "construct a small world" is here:
https://github.com/allenai/allennlp/blob/4f8e04e6cdfbc8707707ac978a964777459c161e/tests/models/encoder_decoders/wikitables_semantic_parser_test.py#L181-L224

Again this test looks long in number of lines, but there are no loops, and there's only one method that's called - the rest is just instantiating some objects that we use for the test, and those objects happen to be longish lists, so the test takes a bunch of lines.

A final example: https://github.com/allenai/allennlp/blob/4f8e04e6cdfbc8707707ac978a964777459c161e/tests/nn/util_test.py#L71-L181

This test is checking a softmax function, which does something very similar to what you're doing.  The test just instantiates a tensor with made up numbers, calculates the expected values by hand, and assert that they end up correct.  No loops, no complex logic, just some simple examples that you can look at and know that they look right (there are also probably way more checks there than are necessary, but that's for historical reasons).

So, what I would recommend for this test is to construct a world with just a few entities, make up a `linking_scores` tensor with random numbers that are just given explicitly in a matrix (but _not_ with `torch.rand()`), and call `._get_linking_probabilities`, and have a check on the exact tensor that's returned.